app/controllers/user_controller.rb

```diff
@@ -0,0 +1,56 @@
+class UserController < ApplicationController
+
+  get '/' do
+    erb :index
+  end
+
+  get '/login' do
+    if !User.is_logged_in?(session)
+      erb :'/users/login'
+    else
+      redirect to "/tweets"
+    end
+  end
+
+  post '/login' do
+    @user = User.find_by(username: params[:username])
+    if @user && @user.authenticate(params[:password])
+      session[:user_id] = @user.id
+      redirect '/tweets'
+    else
+      redirect '/login'
+    end
+  end
+
+  get '/signup' do
+    if User.is_logged_in?(session)
+      redirect to '/tweets'
+    else
+      erb :'users/create_user'
+    end
+  end
+
+  post '/signup' do
+    if params[:username] == "" || params[:password] == "" || params[:email] == ""
```

Comment:
I think you could use [ActiveRecord validations](https://learn.co/tracks/full-stack-web-development/sinatra/activerecord/user-authentication-in-sinatra) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.